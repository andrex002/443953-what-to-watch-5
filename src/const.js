const filmsCount = {
  PER_STEP: 8,
  SIMILAR: 4
};

const FilmTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

const VIDEO_PLAY_TIMEOUT = 1000;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const StatusIsFavorite = {
  ADD: 1,
  REMOVE: 0
};

export {FilmTabs, filmsCount, VIDEO_PLAY_TIMEOUT, AuthorizationStatus, StatusIsFavorite};
