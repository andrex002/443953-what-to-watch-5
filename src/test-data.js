export const film = {
  "id": 1,
  "title": "The Grand Budapest Hotel",
  "image": "img/the-grand-budapest-hotel-poster.jpg",
  "previewImage": "img/the-grand-budapest-hotel.jpg",
  "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
  "bgColor": "#ffffff",
  "srcVideo": "https://some-link",
  "video": "https://some-link",
  "previewVideo": "https://some-link",
  "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
  "rating": {
    "score": 8.9,
    "count": 240
  },
  "director": "Wes Andreson",
  "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
  "duration": 99,
  "genre": "Comedy",
  "year": 2014,
  "isFavorite": false
};

export const films = [
  {
    "id": 1,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  },
  {
    "id": 2,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  }
];

export const comment = {
  "id": 1,
  "userId": 4,
  "userName": "Kate Muir",
  "rating": 8.9,
  "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  "date": "2019-05-08T14:13:56.569Z"
};

export const comments = [
  {
    "id": 1,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
  {
    "id": 2,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
];

export const defaultState = {
  DATA: {
    allFilms: films,
    filteredFilms: films,
    promoFilm: film,
    currentFilm: film,
    favoriteFilms: films,
    comments: comments,
    numberFilmsShown: 8,
    isCommentSending: false,
    isCommentSendError: false
  },
  FILTER: {
    activeGenre: `All genres`,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    name: `userName`,
    avatar: `avatar.jpg`
  }
};

export const noop = () => {};