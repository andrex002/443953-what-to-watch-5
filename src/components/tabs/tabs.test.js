import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs";
import {films, noop} from "../../test-data";
import {FilmTabs} from "../../const";

jest.mock(`../film-page-overview/film-page-overview`, () => `FilmPageOverview`);
jest.mock(`../film-page-details/film-page-details`, () => `FilmPageDetails`);
jest.mock(`../film-page-reviews/film-page-reviews`, () => `FilmPageReviews`);

describe(`Should Tabs render correctly`, () => {
  it(`Active tab Overview`, () => {
    const tree = renderer.create(
        <Tabs
          film={films[0]}
          activeTab={FilmTabs.OVERVIEW}
          handleActiveTab={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Active tab Details`, () => {
    const tree = renderer.create(
        <Tabs
          film={films[0]}
          activeTab={FilmTabs.DETAILS}
          handleActiveTab={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Active tab Reviews`, () => {
    const tree = renderer.create(
        <Tabs
          film={films[0]}
          activeTab={FilmTabs.REVIEWS}
          handleActiveTab={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
