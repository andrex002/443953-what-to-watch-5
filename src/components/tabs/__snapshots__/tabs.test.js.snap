// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly Active tab Details 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <FilmPageDetails
    actors={
      Array [
        "Bill Murray",
        "Edward Norton",
        "Jude Law",
        "Willem Dafoe",
        "Saoirse Ronan",
      ]
    }
    director="Wes Andreson"
    duration={99}
    genre="Comedy"
    year={2014}
  />
</div>
`;

exports[`Should Tabs render correctly Active tab Overview 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <FilmPageOverview
    actors={
      Array [
        "Bill Murray",
        "Edward Norton",
        "Jude Law",
        "Willem Dafoe",
        "Saoirse Ronan",
      ]
    }
    description="In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege."
    director="Wes Andreson"
    rating={
      Object {
        "count": 240,
        "score": 8.9,
      }
    }
  />
</div>
`;

exports[`Should Tabs render correctly Active tab Reviews 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <FilmPageReviews
    filmId={1}
  />
</div>
`;
